cookiecutter/prompt.py

```diff
@@ -217,6 +219,37 @@ def render_variable(env, raw, cookiecutter_dict):
     return template.render(cookiecutter=cookiecutter_dict)
 
 
+def _extract_prompts(context: dict) -> dict:
+    """Extract __prompts__ from context dict."""
+    key = '__prompts__'
+    prompts = {}
+    if key in context['cookiecutter']:
+        prompts = context['cookiecutter'][key]
+        del context['cookiecutter'][key]
+    return prompts
```

Comment:
This duplicates `dict.pop()`. Please remove this function and change all call sites to follow this style:

```python
# Before
_extract_prompts(context=context)

# After
context['cookiecutter'].pop('__prompts__', {})
```